src/runtime/security/negotiation_service.cpp

```diff
@@ -67,9 +67,10 @@ void negotiation_service::on_negotiation_request(negotiation_rpc rpc)
             static_cast<server_negotiation *>(_negotiations[rpc.dsn_request()->io_session].get());
     }
 
-    dassert(srv_negotiation != nullptr,
-            "negotiation is null for msg: {}",
-            rpc.dsn_request()->rpc_code().to_string());
+    if (nullptr == srv_negotiation) {
+        ddebug("negotiation is null for msg: {}", rpc.dsn_request()->rpc_code().to_string());
```

Comment:
```suggestion
        derror("negotiation is null for msg: {}", rpc.dsn_request()->rpc_code().to_string());
```

This case appears when the client is incorrectly sending requests before negotiation completes. In such a case, I think logging in error-level could help us find the cause.

Reply:
Could we just reply the unnegotiated messages using ERR_UNNEGOTIATED, without shutting down the session?

Reply:
I think shut down the session is the best practise. Because if we don't close it, the client can also send message to server. It is waste of network bandwidth